src/main/java/subway/line/LineService.java

```diff
@@ -63,4 +74,62 @@ private StationsAtLine getStationsAtLine(Long upStationId, Long downStationId) {
   private Line getLineById(Long id) {
     return lineRepository.findById(id).orElseThrow(()-> new SubwayException(NOT_FOUND));
   }
+
+  @Transactional
+  public void appendLineSection(Long id, LineSectionAppendRequest lineSectionAppendRequest) {
+    Line line = getLineById(id);
+    List<LineSection> lineSections = getLineSectionsByLineId(id);
+    long index = lineSections.stream().map(item -> item.getIndex()).mapToLong(Long::longValue).max().orElse(1L);
+    Set<Long> lineSectionStations = getUniqueStationIds(lineSections);
+    if(isReqUpstationIdNotEqualsIndex(lineSectionAppendRequest, index) || isReqStationIdExist(lineSectionStations, lineSectionAppendRequest)){
+      throw new SubwayException(BAD_REQUEST);
+    }
+
+    lineSectionRepository.save(createNewLineSection(line, index, lineSectionAppendRequest));
+  }
+
+  @Transactional
+  public void deleteLineSection(Long lineId, Long stationId) {
+    List<LineSection> lineSection = lineSectionRepository.findAllByLineId(lineId);
+
+    LineSection lastLineSection = lineSection.stream().max((a, b) -> (int) (a.getIndex() - b.getIndex()))
+      .orElseThrow(() -> new SubwayException(NOT_FOUND));
+
+    if (lastLineSection.getDownStation().getId() != stationId || lineSection.size() <= 1) {
+      throw new SubwayException(BAD_REQUEST);
+    }
+
+    lineSectionRepository.deleteByLineIdAndStationId(stationId);
+  }
+
+  private List<LineSection> getLineSectionsByLineId(Long lineId) {
+    return lineSectionRepository.findAllByLineId(lineId);
+  }
+
+
+  private Set<Long> getUniqueStationIds(List<LineSection> lineSections) {
+    Set<Long> lineSectionStations = new HashSet<>();
+    lineSections.forEach(item -> {
+      lineSectionStations.add(item.getUpStation().getId());
+      lineSectionStations.add(item.getDownStation().getId());
+    });
+    return lineSectionStations;
+  }
+
+  private boolean isReqUpstationIdNotEqualsIndex(LineSectionAppendRequest lineSectionAppendRequest, Long index) {
+    if (!lineSectionAppendRequest.getUpStationId().equals(index)) return true;
+    return false;
```

Comment:
아래와 같이 줄여서 작성해 볼 수도 있을 것 같아요.
```suggestion
    return !lineSectionAppendRequest.getUpStationId().equals(index);
```